runners/samza/src/main/java/org/apache/beam/runners/samza/adapter/UnboundedSourceSystem.java

```diff
@@ -268,6 +269,7 @@ public void register(SystemStreamPartition ssp, String offset) {
       private final FnWithMetricsWrapper metricsWrapper;
 
       private volatile boolean running;
+      private volatile boolean maxWatermarkReached = false;
```

Comment:
This flag seems a bit cumbersome to understand, as we already have running flag to control the looping.

Reply:
I believe this is change added to support Nexmark. @Zhangyx39  can you comment more on this?

Reply:
This is needed to finish a nexmark test in streaming mode. As @xinyuiscool suggested, we can remove this flag and just use the running flag.